app/models/pipeline_run_stage.rb

```diff
@@ -171,12 +188,11 @@ def prepare_dag(dag_name, attribute_dict, key_s3_params = nil)
                            attribute_dict,
                            pipeline_run.parse_dag_vars)
     self.dag_json = dag.render
-    copy_done_file = "echo done | aws s3 cp - #{sample.sample_output_s3_path}/\\$AWS_BATCH_JOB_ID.#{JOB_SUCCEEDED_FILE_SUFFIX}"
-    upload_dag_json_and_return_job_command(dag_json, dag_s3, dag_name, key_s3_params, copy_done_file)
+    save
+    upload_dag_json(dag_json, dag_s3)
```

Comment:
I also notice that we are uploading the dag json at this point (this is part of the side effect). 
In the previous logic, that file would be uploaded immediately before running it (in this block https://github.com/chanzuckerberg/idseq-web/blob/ca0b34ab1e68ae8c1d3455860c0b2408e531148e/app/models/pipeline_run_stage.rb#L69, which is invoking by reflection one of these methods: https://github.com/chanzuckerberg/idseq-web/blob/ca0b34ab1e68ae8c1d3455860c0b2408e531148e/app/models/pipeline_run_stage.rb#L69)

I think it is a good idea preserving the behavior of uploading this file right before executing it. The existing logic can be modified to upload the contents of this `dag_json` field instead of calculating it (as it currently happens)